resources/session03/wsgi/calculator.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+import re
+import pdb
+
+
+def intro_text():
+    page = """
+    <h1>Online Calculator</h1>
+    <p>This page will perform simple calculations, based on your input.</p>
+    <p>To use the calculator, enter the type of operation, followed by two 
+    numbers, separated by slashes into the page's URL. Accepted operation
+    types are: add, subtract, multiply, divide.</p>
+    <p>For example: http://localhost:8080/<em><strong>add/4/5</strong></em></p>
+    <h3>Your numbers:</h3>
+    <p>{a}, {b}</p>
+    <h3>Answer:</h3>
+    <p>{answer}</p>
+    """
+    return page
+
+
+def strip_path(path):
+    strippedpath = path.lstrip('/')
+    # correct_form = r'^[\w]+/[\d]+/[\d]+$'
+    # match = re.match(correct_form, strippedpath)
+    # if not match:
+    #     raise NameError
+    try:
+        operation, a, b = strippedpath.split('/')
+        return operation, a, b
+    except Exception as e:
+        raise e
+
+
+# def incorrect_syntax():
+#     answer = "Oops! The calculator didn't understand what you entered in the URL."
+#     a = "--"
+#     b = "--"
+#     return answer, a, b
+
+
+def perform_calculation(operation, a, b):
+    num1 = float(a)
+    num2 = float(b)
+    try:
+        if operation == "add":
+            answer = num1 + num2
+        elif operation == "subtract":
+            answer = num1 - num2
+        elif operation == "multiply":
+            answer = num1 * num2
+        elif operation == "divide":
+            try:
+                answer = num1 / num2
+            except ZeroDivisionError:
+                return "No Answer. Cannot divide by zero!"
+        else:
+            answer = "Unknown calculation type."
+        return str(answer)
+    except Exception as e:
+        return str(e)
```

Comment:
notice that by catching any possible exception at this point, you are effectively preventing your application code from handling the exception by returning a `500 Internal Server Error` response.  This function will never ever throw an error, which is probably not what you really want.  In general a function like this should return an appropriate answer value or raise an error, and unexpected exceptions should be handled outside the scope of the function.